### PR TITLE
Style optimized tiles for offline regions

### PR DESCRIPTION
**Pull request by brunoabinader**
This PR adds a new parameter for the `mbgl-offline` tool to enable requesting style-optimized tiles when downloading offline regions.

The extra query gets appended to the tile resource URL, and works as intended according to this [unit test](https://github.com/mapbox/mapbox-gl-native/blob/master/test/util/mapbox.test.cpp#L19-L21).

/cc @tmpsantos @ivovandongen @kkaefer

### CODE REVIEW

File: platform/default/mbgl/storage/offline_download.cpp [hunk 1]
```diff
@@ -219,19 +223,26 @@ void OfflineDownload::activateDownload() {
...
+    ensureResource(Resource::style(styleURL, Resource::Priority::Low),
+                   [&, styleURL](Response styleResponse) {
         status.requiredResourceCountIsPrecise = true;
 
         style::Parser parser;
         parser.parse(*styleResponse.data);
 
+        std::string queryExtras;
+        if (static_cast<uint8_t>(options) & static_cast<uint8_t>(OfflineRegionDownloadOptions::StyleOptimizedTiles)) {
+            queryExtras = "?style=" + styleURL + "@" + parser.modified;
```
Comment (kkaefer) [inline platform/default/mbgl/storage/offline_download.cpp]: Is `@<modified>` an official API?

File: src/mbgl/style/parser.cpp [hunk 2]
```diff
@@ -113,6 +113,24 @@ StyleParseResult Parser::parse(const std::string& json) {
...
     }
 
+    if (document.HasMember("created")) {
+        const JSValue& value = document["created"];
+        if (value.IsString()) {
+            created = { value.GetString(), value.GetStringLength() };
+        } else if (value.IsNumber()) {
+            created = std::to_string(value.GetInt());
+        }
+    }
```
Comment (kkaefer) [inline src/mbgl/style/parser.cpp]: `created` is not used anywhere

File: platform/default/mbgl/storage/offline_download.cpp (hunk 1 above)
Comment (brunoabinader) [reply platform/default/mbgl/storage/offline_download.cpp]: Yes - it is described in the "Request style-optimized tiles" section of the [API documentation](https://www.mapbox.com/api-documentation/#retrieve-tiles) as _timestamp_:
> The  style parameter is broken into two parts, the style ID and the style's recently edited  timestamp . The timestamp parameter comes from the style JSON's modified property, which is included with any style created with Mapbox Studio.

**Comment by brunoabinader** [comment]
> Great! But how are the build changes related to this PR?

I didn't want to spend resources on creating a separate PR just for these. These changes removes build script redundancies and enables building the project on Visual Studio Code without any separate CMake argument required.

**Comment by brunoabinader** [comment]
> I'm not sure this is a good idea. Requesting style-optimized tiles without changing the tile URL template means that we'll end up with stripped/optimized tiles, while other styles might expect full tiles. Instead, I think we should explore further along #10864 and split up layers in the database so we don't have the duality of an optimized tile and a non-optimized tile anymore.

I agree with further exploring #10864, but I believe they solve different problems.

Users who want to use style-optimized tiles should be aware of the drawbacks when dealing with multiple styles, but instead want to benefit from the considerably smaller cache databse size when using only a single style.

File: CMakeLists.txt [hunk 3]
```diff
@@ -21,7 +22,7 @@ set(CMAKE_VISIBILITY_INLINES_HIDDEN 1)
 
 set(CMAKE_OSX_DEPLOYMENT_TARGET 10.11)
 
-option(WITH_CXX11ABI "Use the C++11 ABI" OFF)
+option(WITH_CXX11ABI "Use the C++11 ABI" ON)
```
Comment (kkaefer) [inline CMakeLists.txt]: Why did you change the default?

Comment (brunoabinader) [reply CMakeLists.txt]: It's been long since our development machines and CI have departed from previous C++ implementations using the old ABI.

**Comment by brunoabinader** [comment]
Abandoning this as @springmeyer helped me realize the same expected result can be achieved by appending `?optimize=true` to the style URL. I'll forward the build changes to a separate PR.
